src/atomate2/common/flows/gruneisen.py

```diff
@@ -0,0 +1,186 @@
+"""Flows for calculating Grueneisen-Parameters."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+from jobflow import Flow, Maker
+
+from atomate2.common.jobs.gruneisen import (
+    compute_gruneisen_param,
+    shrink_expand_structure,
+)
+
+if TYPE_CHECKING:
+    from pymatgen.core.structure import Structure
+
+    from atomate2.aims.jobs.base import BaseAimsMaker
+    from atomate2.common.flows.phonons import BasePhononMaker
+    from atomate2.forcefields.jobs import ForceFieldRelaxMaker
+    from atomate2.vasp.jobs.base import BaseVaspMaker
+
+
+@dataclass
+class BaseGruneisenMaker(Maker, ABC):
+    """
+    Maker to calculate Grueneisen parameters with DFT/force field code and Phonopy.
+
+    Calculate the harmonic phonons of a material for and compute Grueneisen parameters.
+    Initially, a tight structural relaxation is performed to obtain a structure without
+    forces on the atoms. The optimized structure (ground state) is further expanded and
+    shrunk by 1 % of its volume. Subsequently, supercells with one displaced atom are
+    generated for all the three structures (ground state, expanded and shrunk volume)
+    and accurate forces are computed for these structures. With the help of phonopy,
+    these forces are then converted into a dynamical matrix. This dynamical matrix of
+    three structures is then used as input phonopy Grueneisen api to compute Grueneisen
+    parameters are computed.
+
+
+    Parameters
+    ----------
+    name : str
+        Name of the flows produced by this maker.
+    bulk_relax_maker: .ForceFieldRelaxMaker, .BaseAimsMaker, .BaseVaspMaker, or None
+        A maker to perform an initial tight relaxation on the bulk.
+    code: str
+        determines the dft or force field code.
+    const_vol_relax_maker: .ForceFieldRelaxMaker, .BaseAimsMaker,
+        .BaseVaspMaker, or None. A maker to perform a tight relaxation
+        on the expanded and shrunk structures at constant volume.
+    kpath_scheme: str
+        scheme to generate kpoints. Please be aware that
+        you can only use seekpath with any kind of cell
+        Otherwise, please use the standard primitive structure
+        Available schemes are:
+        "seekpath", "hinuma", "setyawan_curtarolo", "latimer_munro".
+        "seekpath" and "hinuma" are the same definition but
+        seekpath can be used with any kind of unit cell as
+        it relies on phonopy to handle the relationship
+        to the primitive cell and not pymatgen
+    mesh: tuple
```

Comment:
this should not be a mesh .it should be a density/or a reciprocal distance similar to the phonon workflow.

Reply:
Do you mean the arg name?

Reply:
No, i rwally mean the input. Will fix this. Don't worry. Just leaving comments.

Reply:
ah, just got reminded about this part here....

Reply:
added new density option and switched to gamma centered meshes. Will need to check if this is a good idea for the grüneisen parameter